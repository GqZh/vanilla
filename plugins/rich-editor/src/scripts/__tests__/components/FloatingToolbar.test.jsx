import React from "react";
import { mount, shallow } from "enzyme";
import FloatingToolbar from "../../components/FloatingToolbar";
import Quill from "quill/core";

jest.mock("quill/core");

describe("<FloatingToolbar />", () => {
    const quill = new Quill();
    quill.options = {
        modules: {
            keyboard: {
                bindings: {},
            },
        },
    };
    quill.scroll = {
        descendants: jest.fn(() => []),
    };

    const toolbar = shallow(
        <FloatingToolbar quill={quill} forceVisibility={"visible"}>
            <div>Hello World</div>
        </FloatingToolbar>
    );

    /** @var {FloatingToolbar} */
    const instance = toolbar.instance();

    it("matches snapshot", () => {
        expect(toolbar).toMatchSnapshot();
    });
});
