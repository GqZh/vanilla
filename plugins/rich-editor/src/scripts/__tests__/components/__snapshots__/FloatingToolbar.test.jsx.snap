// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingToolbar /> matches snapshot 1`] = `
<div
  className="richEditor-inlineMenu "
  style={
    Object {
      "position": "absolute",
      "visibility": "hidden",
    }
  }
>
  <div>
    Hello World
  </div>
  <div
    className="richEditor-nubPosition"
    style={Object {}}
  >
    <div
      className="richEditor-nub"
    />
  </div>
</div>
`;
